### ISSUE TITLE
UX: The HTML Editor Manager doesn't fit in the Personabar

### ISSUE THREAD

**Issue by OllyHodgson** [dnnsoftware/Dnn.AdminExperience#78]
## Description of problem
The CKEditor configuration is obviosuly designed to be hosted on its own full-size page (a lot like the current File Manager). It's being loaded into an 800px wide iframe where it's very squashed. Users sometimes have to "double-scroll" (scroll the iframe, and an overflow box within it) to reach some very important functionality.

## Description of solution
Rebuild the management interface as a proper PersonaBar extension, so it fits and works correctly on this page. Ideas for saving space include making the sidebar collapsible, and putting the icon legend into help bubble, as per other "help" items. If possible, make it "responsive", so it adapts to fit a variety of screen sizes.

## Description of alternatives considered
1. Restyle the current interface to work a bit better in the constrained space. This feels a bit like 'polishing a turd' though.
2. Make the PersonaBar wider for this page (like the current Content > Pages interface). This feels like a short term solution, and doesn't fit well with the goal of making PersonaBar work nicely on mobile.

## Additional context
The HTML editor config manager is easy to use. 
It's styled to match the rest of the PersonaBar. 
It physically fits into the space provided for it.

## Additional context
I'm guessing this would need to accommodate HTML editor providers other than the default?

## Affected version

* [x] 9.2.2
* [x] 9.2.1
* [x] 9.2
* [x] 9.1.1
* [x] 9.1
* [x] 9.0

**Comment by valadas** [comment]
@dnnsoftware/tag what are the opinions on this. 

Being provider based, we need to account for different modules having different UIs for their settings right?
One quick way would be for that persona bar link to just open a settings page for the editor instead of being inside the Persona Bar.

Making the settings view responsive I think would be on the actual provider repository.

And if we make it a persona bar module then that would also most likely be on the provider repository, but then we need to be able to find out if the provider uses a persona bar module or and older webforms/mvc module view for its settings...

Toughts?

**Comment by valadas** [comment]
One quick fix I think could be making the panel larger like it is for pages, not perfect though but there would be multiple places to correctly do this.

**Comment by OllyHodgson** [comment]
> Making the settings view responsive I think would be on the actual provider repository.

I agree completely. Is this particular item a bug for https://github.com/w8tcha/dnnckeditor or https://github.com/DNN-Connect/CKEditorProvider?

If we're going to continue to iframe external settings pages like this, we could look into exposing an API to allow that page to resize the iframe (vertically) us when their content size changes.

**Comment by valadas** [comment]
Well for the default provider it would be in the Dnn-Connect one, but since this is provider based, then each optional html provider needs to ensure their settings UI is responsive. Us here increasing the panel size is a big workaround and only a desktop solution. Ultimatelly making the persona bar responsive this workaround would not be of much help to the providers, they need to be responsive themselves and not by the page size but the div that contains it (javascript involved here).

**Comment by sleupold** [comment]
this is an admin task, and if it works properly on the desktop, that would be a great improvement.
